pkg/storage/swift/swift.go

```diff
@@ -343,14 +343,27 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 		return err
 	}
 
+	infra, err := util.GetInfrastructure(d.Listers)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster infrastructure info: %v", err)
+	}
+
 	// Generate new container name if it wasn't provided.
-	// The name has a prefix "image_registry_", which is complemented by 16 capital latin letters
-	// Example of a generated name: image_registry_FHEIBGDDGBLWPXFR
+	// The name has cluster ID as a prefix plus "image-registry" suffix, which is complemented
+	// by 8 capital latin letters
+	// Example of a generated name: user-298xw-image-registry-FHEIBGDD
 	if cr.Spec.Storage.Swift.Container == "" {
-		cr.Spec.Storage.Swift.Container = generateContainerName("image_registry_")
+		cr.Spec.Storage.Swift.Container = generateContainerName(infra.Status.InfrastructureName)
+	}
+
+	createOps := containers.CreateOpts{
+		Metadata: map[string]string{
+			"Openshiftclusterid": infra.Status.InfrastructureName,
+			"Name":               cr.Spec.Storage.Swift.Container,
+		},
 	}
 
-	_, err = containers.Create(client, cr.Spec.Storage.Swift.Container, containers.CreateOpts{}).Extract()
+	_, err = containers.Create(client, cr.Spec.Storage.Swift.Container, createOps).Extract()
```

Comment:
Can you wrap this create in a loop that will try again with a newly generated name if the name is already taken? (set a limit of some kind to keep it from looping forever)